diff --git a/tools/depends/configure.ac b/tools/depends/configure.ac
--- a/tools/depends/configure.ac
+++ b/tools/depends/configure.ac
@@ -266,0 +267,10 @@
+  x86_64*-*-freebsd*)
+    use_toolchain="${use_toolchain:-/opt/OpenOrbis/PS4Toolchain}"
+    platform_cc=clang
+    platform_cxx=clang++
+    platform_tool_prefix=llvm-
+    platform_additional_path="$use_toolchain/bin/linux"
+    if test "$use_linker_explicit" != "yes"; then
+      use_linker=ld.lld
+    fi
+  ;;
@@ -415,0 +416,13 @@
+  x86_64*-*-freebsd*)
+    if test "x$use_cpu" = "xauto"; then
+      use_cpu="x86_64"
+    fi
+    meson_cpu="x86_64"
+    meson_system="freebsd"
+    platform_os="freebsd"
+    target_platform="ps4"
+    platform_cflags="--target=x86_64-pc-freebsd12-elf -fPIC -funwind-tables -isysroot $use_toolchain -isystem $use_toolchain/include"
+    platform_cxxflags="--target=x86_64-pc-freebsd12-elf -fPIC -funwind-tables -isysroot $use_toolchain -isystem $use_toolchain/include/c++/v1 -isystem $use_toolchain/include"
+    optimize_flags="-O2"
+    platform_ldflags="--target=x86_64-pc-freebsd12-elf -fuse-ld=lld -nostdlib -pie -Wl,--script=$use_toolchain/link.x -Wl,--eh-frame-hdr -L$use_toolchain/lib $use_toolchain/lib/crt1.o -lc -lkernel -lc++"
+  ;;
diff --git a/tools/depends/configure.ac b/tools/depends/configure.ac
--- a/tools/depends/configure.ac
+++ b/tools/depends/configure.ac
@@ -585,6 +585,8 @@
 case $use_platform in
+  ps4)
+    ;;
   ios|ios-simulator|tvos|tvos-simulator|macos)
     ;;
   webos)
     ;;
   wasm)
     ;;
   auto)
     ;;
   *)
     AC_MSG_ERROR(unsupported platform ($use_platform))
 esac

diff --git a/tools/depends/m4/xbmc_arch.m4 b/tools/depends/m4/xbmc_arch.m4
--- a/tools/depends/m4/xbmc_arch.m4
+++ b/tools/depends/m4/xbmc_arch.m4
@@ -46,6 +46,9 @@
   i386-*-freebsd*)
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_FREEBSD")
      ;;
+  x86_64-*-freebsd*)
+     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_FREEBSD")
+     ;;
   amd64-*-freebsd*)
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_FREEBSD")
      ;;
 
diff --git a/tools/depends/target/Toolchain.cmake.in b/tools/depends/target/Toolchain.cmake.in
--- a/tools/depends/target/Toolchain.cmake.in
+++ b/tools/depends/target/Toolchain.cmake.in
@@ -36,0 +37,5 @@
+elseif(OS STREQUAL freebsd)
+  set(CMAKE_SYSTEM_NAME FreeBSD)
+  set(CORE_SYSTEM_NAME ps4)
+  set(CORE_PLATFORM_NAME ps4)
+  set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
