diff --git a/tools/depends/target/openssl/Makefile b/tools/depends/target/openssl/Makefile
--- a/tools/depends/target/openssl/Makefile
+++ b/tools/depends/target/openssl/Makefile
@@ -1,3 +1,3 @@
 include ../../Makefile.include OPENSSL-VERSION ../../download-files.include
-DEPS = ../../Makefile.include Makefile OPENSSL-VERSION ../../download-files.include \
+DEPS = ../../Makefile.include Makefile OPENSSL-VERSION ../../download-files.include 0001-openorbis-in6-addr.patch 0002-openorbis-endian-header.patch \
                         001-android-getauxvalrevert.patch 16-kodi.conf
@@ -71,7 +71,8 @@
 ifeq ($(OS),android)
 	cd $(PLATFORM); patch -p1 -i ../001-android-getauxvalrevert.patch
 endif
 	cd $(PLATFORM); patch -p1 -i ../0001-openorbis-in6-addr.patch
+	cd $(PLATFORM); patch -p1 -i ../0002-openorbis-endian-header.patch
 	cd $(PLATFORM); cp ../16-kodi.conf ./Configurations/
 	cd $(PLATFORM); $(CONFIGURE)
 sed -ie "s|PROGRAMS=|PROGRAMS=#|" "$(PLATFORM)/Makefile";
